### ISSUE TITLE
[0.3] Strange behaviour with highlights

### ISSUE THREAD

**Issue by bdauria** [futurepress/epub.js#661]
I've noticed a strange behaviour using the highlight feature. It seems like as soon as the highlight() function is called, the view is growing in width indefinitely, which makes the content located after the highlighted section unreachable. The following gif illustrates the problem: 

![highlight](https://user-images.githubusercontent.com/8367233/30764101-c964826e-9fe9-11e7-8fbf-4140af5b373d.gif)

In this example, I'm simply using the snippet taken from the examples:

```
    rendition.on('selected', (cfiRange, contents) => {
      rendition.annotations.highlight(cfiRange, {}, e => {
        console.log('highlight clicked', e.target);
      });
      contents.window.getSelection().removeAllRanges();
    });
```
The same thing seem to happen with underline. 

any idea where that might be coming from ?

**Comment by bdauria** [comment]
It looks like the issue comes from this: 
https://github.com/fchasen/marks/commit/14ca703be91157716769c2454aa3dc9b6d867684

I've tried 
```
function coords(el) {
    var rect = el.getBoundingClientRect();

    return {
        top: rect.top + el.ownerDocument.body.scrollTop,
        left: rect.left + el.ownerDocument.body.scrollLeft,
        height: rect.height,
        width: rect.width
    };
}


function setCoords(el, coords) {
    el.style.top = `${coords.top}px`;
    el.style.left = `${coords.left}px`;
    el.style.height = `${coords.height}px`;
    el.style.width = `${coords.width}px`;
}

```
which seems to fix it.

**Comment by fchasen** [comment]
Would be ideal to move the marks pane outside of the iframe and into the view div. I'll look into if there is anything blocking this.

**Comment by bdauria** [comment]
Yes, that would probably be the solution. Because even with a fix size, I've noticed that the marks overlay triggers unwanted resize hooks which creates blank pages.

**Comment by fchasen** [comment]
Moved the marks from the iframe to the view on v0.3 branch, will publish it to npm shortly.

**Comment by fchasen** [comment]
Published in > 0.3.53
